src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -61,45 +50,33 @@ public User get(String email, String password) {
     try {
       Response authResponse = client.newCall(authRequest).execute();
       String authResponseData = authResponse.body().string();
+      JSONObject authResponseJson = new JSONObject(authResponseData);
 
       if (!authResponse.isSuccessful()) {
-        System.out.println("Authentication failed: " + authResponseData);
-        return null;
+        String failureMessage = authResponseJson.getJSONObject("error").getString("message");
+        throw new RuntimeException("Authentication failed: " + failureMessage);
       }
 
-      JSONObject authResponseJson = new JSONObject(authResponseData);
+      if (!authResponse.isSuccessful()) {
+        String failureMessage = authResponseJson.getJSONObject("error").getString("message");
+        throw new RuntimeException("Authentication failed: " + failureMessage);
```

Comment:
We should map common Firebase error messages to something nicer to display for the user.